webpack/assets/javascripts/react_app/common/I18n.js

```diff
@@ -1,6 +1,40 @@
 import Jed from 'jed';
+import { addLocaleData } from 'react-intl';
 import { deprecateObjectProperty } from '../../foreman_tools';
 
+const runningInPhantomJS = () => (window._phantom !== undefined);
+
+class IntlLoader {
+  constructor(locale, timezone) {
+    this.fallbackIntl = !global.Intl || runningInPhantomJS();
+
+    [this.locale] = locale.split('-');
+    this.timezone = this.fallbackIntl ? 'UTC' : timezone;
+    this.loaded = false;
+    this.ready = this.init();
```

Comment:
I am curious if both `loaded` and `ready` are needed. Please disregard if I missed something, but `loaded` is initially set to `false` and flipped when the async actions finish, while `ready` contains the promise returned by calling `init()`. 

`loaded` is used only in `constructor` of `I18nProviderWrapperFactory`, where we wait for the `ready` promise to resolve if `!loaded`. So I was thinking you could `return true` from `init()` and always wait for the promise to resolve - if the `init()` call already finished, then promise resolves immediately and if not, we wait for it just like before.

This [diff](https://github.com/xprazak2/foreman/commit/cbcdbb004ab57e29cae64a0f3682d86f9d88902f) seemed to work...

Reply:
I think it's leftover from earlier stages of development. What you write makes sense. I'll test your patch and merge it into my branch if it works. Thanks!